.buildkite/scripts/serverless/create_deploy_tag/release_wizard_messaging.ts

```diff
@@ -0,0 +1,374 @@
+      return result !== false;
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  } else {
+    return true;
+  }
+}
+
+async function sendReleaseSlackAnnouncement({
+  targetCommitData,
+  currentCommitSha,
+  deployTag,
+}: {
+  targetCommitData: GithubCommitType;
+  currentCommitSha: string | undefined | null;
+  deployTag: string;
+}) {
+  const textBlock = (...str: string[]) => ({ type: 'mrkdwn', text: str.join('\n') });
+  const buildShortname = `kibana-serverless-release #${process.env.BUILDKITE_BUILD_NUMBER}`;
+
+  const isDryRun = process.env.DRY_RUN?.match('(1|true)');
+  const mergedAtDate = targetCommitData.commit?.committer?.date;
+  const mergedAtUtcString = mergedAtDate ? new Date(mergedAtDate).toUTCString() : 'unknown';
+  const targetCommitSha = targetCommitData.sha;
+  const targetCommitShort = targetCommitSha.slice(0, 12);
+  const compareResponse = (
+    await octokit.repos.compareCommits({
+      owner: 'elastic',
+      repo: 'kibana',
+      base: currentCommitSha || 'main',
+      head: targetCommitSha,
+    })
+  ).data;
+  const compareLink = currentCommitSha
+    ? `<${compareResponse.html_url}|${compareResponse.total_commits} new commits>`
+    : 'a new release candidate';
+
+  const mainMessage = [
+    `:ship_it_parrot: Promotion of ${compareLink} to QA has been <${process.env.BUILDKITE_BUILD_URL}|initiated>!\n`,
+    `*Remember:* Promotion to Staging is currently a manual process and will proceed once the build is signed off in QA.\n`,
+  ];
+  if (isDryRun) {
+    mainMessage.unshift(
+      `*:memo:This is a dry run - no commit will actually be promoted. Please ignore!*\n`
+    );
+  } else {
+    mainMessage.push(`cc: @kibana-serverless-promotion-notify`);
+  }
+
+  const linksSection = {
+    'Initiated by': process.env.BUILDKITE_BUILD_CREATOR || 'unknown',
+    'Pre-release job': `<${process.env.BUILDKITE_BUILD_URL}|${buildShortname}>`,
+    'Git tag': `<https://github.com/elastic/kibana/releases/tag/${deployTag}|${deployTag}>`,
+    Commit: `<https://github.com/elastic/kibana/commit/${targetCommitShort}|${targetCommitShort}>`,
+    'Merged at': mergedAtUtcString,
+  };
+
+  const usefulLinksSection = getUsefulLinks({
+    previousCommitHash: currentCommitSha || 'main',
+    selectedCommitHash: targetCommitSha,
+  });
+
+  sendSlackMessage({
+    blocks: [
+      {
+        type: 'section',
+        text: textBlock(...mainMessage),
+      },
+      {
+        type: 'section',
+        fields: Object.entries(linksSection).map(([name, link]) => textBlock(`*${name}*:`, link)),
+      },
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text:
+            '*Useful links:*\n\n' +
+            Object.entries(usefulLinksSection)
+              .map(([name, link]) => ` • <${link}|${name}>`)
+              .join('\n'),
+        },
+      },
+    ],
+  }).catch((error) => {
+    console.error("Couldn't send slack message.", error);
```

Comment:
We may need to redact the axios error, I'm not sure how to test it though.  Can you help?  Changes LGTM otherwise.

side note for me:  eslint/prettier is not configured in .buildkite.

Reply:
good point, this was a recent one I didn't think about! I'll hide it